site/src/utils/portForward.ts

```diff
@@ -56,9 +58,28 @@ export const openMaybePortForwardedURL = (
         agentName,
         workspaceName,
         username,
+        url.protocol.replace(":", "") as WorkspaceAgentPortShareProtocol,
       ) + url.pathname,
     );
   } catch (ex) {
     open(uri);
   }
 };
+
+export const saveWorkspaceListeningPortsProtocol = (
+  workspaceID: string,
+  protocol: WorkspaceAgentPortShareProtocol,
+) => {
+  localStorage.setItem(
+    `listening-ports-protocol-workspace-${workspaceID}`,
+    protocol,
+  );
+};
+
+export const getWorkspaceListeningPortsProtocol = (
+  workspaceID: string,
+): WorkspaceAgentPortShareProtocol => {
+  return (localStorage.getItem(
+    `listening-ports-protocol-workspace-${workspaceID}`,
+  ) || "http") as WorkspaceAgentPortShareProtocol;
+};
```

Comment:
I'm always wary of using `localStorage` for stuff like this. Can you share some of your rationale?

Reply:
I am too. We have had many group discussions (dean, jon, kyle, ben) around a few of the PRs coming out of Shared Ports about these protocol selection features. I've suggested keeping this state and other UX state in the DB in the past, but the group thought that putting this button in local storage is better than making a new table to track this stuff. We want an MVP that is feature complete, but can still iterate on later as we get customer feedback, and this was the solution we landed on for now.

Reply:
Gotcha. If it changes the protocol of _all_ ports though, is it really "UI state"?

Reply:
I promise you I have brought up many different discussions about doing things with this on the backend but the conclusion the group came to was client side storage for now and that we can iterate on the design once we get feedback.